src/modeliteration.js

```diff
@@ -284,24 +284,20 @@ define([
     }
 
     function updateDataSource(mug, value, previous) {
-        var value_src = value && value.instance && value.instance.src,
-            prev_src = previous && previous.instance && previous.instance.src;
-        if (value_src !== prev_src) {
-            var ref = mug.ufid + ".dataSource.instance";
-            if (prev_src) {
-                mug.form.dropInstanceReference(prev_src, ref);
-            }
-            if (value_src) {
-                var instanceId = mug.form.addInstanceIfNotExists(value.instance, ref);
-                if (instanceId !== value.instance.id) {
-                    // is it too magical to replace the instance id in the query?
-                    // there might be edge cases where a user is entering a
-                    // custom instance and query and does not want this to
-                    // happen
-                    value.idsQuery = value.idsQuery.replace(
-                        instanceRegexp, "instance('" + instanceId + "')");
-                    mug.p.dataSource = value; // fire change handler
```

Comment:
Correct, this code is triggered by/part of a change handler already. Also, I think this never caused anything to happen anyway because the change handler only fires if the value actually changes (`value !== old`), which I'm pretty sure is never true.

I also manually tested this to verify that the UI acts the way we expect.